src/monalisten/_sse.py

```
@@ -0,0 +1,30 @@
+from __future__ import annotations
+
+import asyncio
+import math
+from itertools import count
+from typing import TYPE_CHECKING
+
+import httpx
+from httpx_sse import ServerSentEvent, aconnect_sse
+
+if TYPE_CHECKING:
+    from collections.abc import AsyncIterator
+
+
+async def aiter_sse_retrying(
+    client: httpx.AsyncClient, method: str, url: str
+) -> AsyncIterator[ServerSentEvent]:
+    last_event_id: str | None = None
+    retry_delay = 0.0
+    for attempt in count():
```

Do you want an infinite retry? Should you not _eventually_ fail? I don't love the idea that this can be silently failing in an infinite loop in the background. To that end, should debug logging be brought into this package? (Just mentioning it, I don't have a strong opinion either way).

---

From some online reading it looks like it's implied the client should *always* attempt to reconnect in the context of SSE, unless explicitly told to stop by being sent a 204 response, which is handled by httpx.